SCE-Development/dev
# Add Auto gRPC Code Generation to SCE

We can now do `sce generate <path to proto> --language <language types>` and have gRPC code be generated for us.

### How to test this
1. Save [`print.proto`](https://raw.githubusercontent.com/SCE-Development/SCE-RPC/master/proto/print.proto) within the `dev` directory.
1. Make a copy of `config/config.example.json` to `config/config.json` and replace the `SERVER_URL` with what i sent in discord github channel
1. run `sce generate --proto <path to print.proto> --langage js py`, you should then get an output like below:
![image](https://user-images.githubusercontent.com/36345325/89504450-de142480-d77c-11ea-8b72-cb39655f891f.png)
1. Open one of the urls and download it. Compare the file you downloaded with the python code [here](https://github.com/SCE-Development/SCE-RPC/tree/master/server/printing) or the js code [here](https://github.com/SCE-Development/SCE-RPC/tree/master/client/printing)
